ordermentum/steveo
# Add Steveo workflow execution implementation (Drama pattern)

#### Description

_NOTE: This is part of a breakdown of the previous "preview" PR that became a little large. Only the Steveo portion of code changes have been ported to this PR_

Adding the implementation of the ["drama pattern"](https://ordermentum.atlassian.net/wiki/spaces/OD/pages/edit-v2/3035758990). This implements the core functionality only, separate PRs will include code for:

- Postgres storage module implementation
- Example standalone application using workflows
- More unit tests for the workflow engine

----
#### Changes

- Change the Registry to handle more than one type of runtime execution type

----

#### Testing Instructions

Ensure Steveo deploys and runs as expected in testing. There are no inherent changes to the pipelines that should affect how they run.
